README.md

```diff
@@ -1,2 +1,288 @@
+---
+# <ins>Theory And Approach
+
+## <ins>Design 
+* **Our First Approach was to design our robot on <font color="green">Solidworks</font>.**
+
+    We designed a Bi-Pedal BOT on the <font color="red">Pantograph Mechanism</font>, which is a mechanical linkage connected in a manner based on parallelograms. 
+
+    The Kinematics of the Bot were Derived for Pentograph design from which the key parameters such as the length of the linkages were able to be modified and optimized during the design process.
+
+    The Hip Joint is defined to rotate in a 180 degree arc which results in the legs moving between Parallel and Anti-Parallel configurations. The angle between the vertical and the legs is shown for both the parallel and anti-parallel configurations and defined as the angles Phi1, and Phi2 as shown in Figure.
+    The trapezoid which represents the body of the robot.
+
+    ![](https://i.imgur.com/hQBeLkl.png)
+
+    In the above Figure Leg extension is defined as the variable L is the vertical distance between the axis of the hip joints and the foot joint. The leg is shown at minimum extension Lmin the anti-parallel configuration and at maximal extension Lmax in the parallel configuration.
+
+    ![](https://i.imgur.com/lHossMi.png)
+
+    The above figure shown when the bot is at an arbitary position.
+    L1,L2,L3 are the length of linkages (System design parameters / variables).
+
+    <font color="green">
+    L1 ---> Length of Upper Leg Segment (From Hip Joint to Knee Joint)
+
+    L2 ---> Length of Lower Leg Segment (From Knee Joint to Foot Joint)
+
+    L3 --> Length from the Hip joint to the Mechanism's axis of Symmetry.
+    </font>
+
+* **Equations Used To Derive the System Design Variables**
+
+    ![](https://i.imgur.com/E1Gui6i.png)
+
+    In Above Image we have equations to calculate the <font color="yellow">$\phi$1, $\phi$2, L1, L2 and L3 </font>.
+
+    ![](https://i.imgur.com/1IjtgOg.png)
+
+    In above Image, it is the General Equation of Length (vertical distance between the axis of the hip joints and the foot joint)
+    Here, G (Gear Ratio) which is taken to be 1 for our Bot.
+    
+    
+    ![](https://i.imgur.com/9046wKp.png)
+
+    The above Graph gives is about change of Leg Extension wrt change in Driver Motor Angle. 
+
+
+    [Link To Python Script Used]()
+    [LInk to Python Script to Plot the Graph]()
+
+* **Specfications of Bot**
+    <font color="red">
+    
+    Lmax = 13 cm
+
+    Lmin = 4 cm
+    
+    L1 = 4.286 cm
+    
+    L2 = 9.236 cm
+
+    L3 = 3.5 cm
+
+    Phi1 = 45
+
+    Phi2 = 15
+
+    </font>
+
+
+
+
+## <ins> Components
+* Bot Parts of Acrylic Material (Laser-Cutted)
+* N20 motor Shaft for Wheels (3D Printed)
+* SRA PCB 
+* N20 motors
+* MG90 Servo Motors
+* Buck Converter (LM2596)
+* Bearings
+* Shafts
+
+> :memo: **Note:** [Link to Components used In Robot and Description](https://github.com/KudoC0nan/Balance_On_Wheels/blob/dev_arsh/Notes/Components.md)
+
+## <ins> PID
```

Comment:
Explain what are you using PID for ie Self Balancing not general explanation about PID. If you want to make PID Notes markdown add graph that is used in your case as well not some general graph. also that markdown does not have an extension so Github is not rendering it.

Reply:
added explanation and made necessary changes